libgdf/src/gdf_table.cuh

```diff
@@ -590,100 +531,35 @@ public:
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        case GDF_INT64:
-          {
-            using col_type = int64_t;
-            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        case GDF_FLOAT32:
-          {
-            using col_type = float;
-            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        case GDF_FLOAT64:
-          {
-            using col_type = double;
-            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        case GDF_DATE32:
-          {
-            using col_type = int32_t;
-            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        case GDF_DATE64:
-          {
-            using col_type = int64_t;
-            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        case GDF_TIMESTAMP:
-          {
-            using col_type = int64_t;
-            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            if(my_elem != other_elem)
-              return false;
-            break;
-          }
-        default:
-          return false;
+
+      bool is_equal = gdf_type_dispatcher(my_col_type, 
+                                          elements_are_equal, 
+                                          0, 
+                                          d_columns_data[i], 
+                                          my_row_index, 
+                                          other.d_columns_data[i], 
+                                          other_row_index);
+
+      if(false == is_equal){
```

Comment:
We're checking that two rows are equal. So you iterate through the columns. If two elements in column `i` are not equal, then we return `false`. Otherwise, keep iterating through the columns. If we get through all the columns without returning false, then we return true.

Reply:
@harrism : If this were using `std::all_of` like it should, this whole function would have been a one-liner, with none of this boolean twiddling.